File: src/test/java/org/storm/engine/example/ImmovableRectEntity.java
package org.storm.engine.example;

import javafx.scene.paint.Color;
import org.storm.core.render.Renderable;
import org.storm.physics.entity.ImmovableEntity;
import org.storm.physics.math.geometry.shapes.AxisAlignedRectangle;
import org.storm.physics.transforms.UnitConvertor;

public class ImmovableRectEntity extends ImmovableEntity {

  public ImmovableRectEntity(double x, double y, double width, double height) {
    super(new AxisAlignedRectangle(x, y, width, height));
  }

  @Override
  public Renderable transform(UnitConvertor unitConvertor) {
    return (gc, x, y) -> {
      gc.setFill(Color.RED);
      super.transform(unitConvertor).render(gc, x, y);
    };
  }

}

File: components/engine/src/test/kotlin/org/storm/engine/example/ImmovableRectEntity.kt
package org.storm.engine.example

import org.storm.core.render.canvas.Canvas
import org.storm.core.render.canvas.Color
import org.storm.physics.entity.ImmovableEntity
import org.storm.physics.math.geometry.shapes.AABB

class ImmovableRectEntity(
    x: Double,
    y: Double,
    width: Double,
    height: Double
) : ImmovableEntity(AABB(x, y, width, height)) {

    override suspend fun render(canvas: Canvas, x: Double, y: Double) {
        canvas.withSettings(canvas.settings.copy(color = Color(255.0, 0.0, 0.0, 1.0))) {
            super.render(canvas, x, y)
        }
    }

}
